src/script/lua_api/l_itemstackmeta.cpp

```diff
@@ -0,0 +1,304 @@
+
+// Exported functions
+
+// garbage collector
+int ItemStackMetaRef::gc_object(lua_State *L) {
+	ItemStackMetaRef *o = *(ItemStackMetaRef **)(lua_touserdata(L, 1));
+	delete o;
+	return 0;
+}
+
+// get_string(self, name)
+int ItemStackMetaRef::l_get_string(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL){
+		lua_pushlstring(L, "", 0);
+		return 1;
+	}
+	std::string str = meta->getString(name);
+	lua_pushlstring(L, str.c_str(), str.size());
+	return 1;
+}
+
+// set_string(self, name, var)
+int ItemStackMetaRef::l_set_string(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+	size_t len = 0;
+	const char *s = lua_tolstring(L, 3, &len);
+	std::string str(s, len);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL || str == meta->getString(name))
+		return 0;
+	meta->setString(name, str);
+	reportMetadataChange(ref);
+	return 0;
+}
+
+// get_int(self, name)
+int ItemStackMetaRef::l_get_int(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = lua_tostring(L, 2);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL){
+		lua_pushnumber(L, 0);
+		return 1;
+	}
+	std::string str = meta->getString(name);
+	lua_pushnumber(L, stoi(str));
+	return 1;
+}
+
+// set_int(self, name, var)
+int ItemStackMetaRef::l_set_int(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = lua_tostring(L, 2);
+	int a = lua_tointeger(L, 3);
+	std::string str = itos(a);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL || str == meta->getString(name))
+		return 0;
+	meta->setString(name, str);
+	reportMetadataChange(ref);
+	return 0;
+}
+
+// get_float(self, name)
+int ItemStackMetaRef::l_get_float(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = lua_tostring(L, 2);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL){
+		lua_pushnumber(L, 0);
+		return 1;
+	}
+	std::string str = meta->getString(name);
+	lua_pushnumber(L, stof(str));
+	return 1;
+}
+
+// set_float(self, name, var)
+int ItemStackMetaRef::l_set_float(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+	std::string name = lua_tostring(L, 2);
+	float a = lua_tonumber(L, 3);
+	std::string str = ftos(a);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if(meta == NULL || str == meta->getString(name))
+		return 0;
+	meta->setString(name, str);
+	reportMetadataChange(ref);
+	return 0;
+}
+
+// to_table(self)
+int ItemStackMetaRef::l_to_table(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	ItemStackMetaRef *ref = checkobject(L, 1);
+
+	ItemStackMetadata *meta = getmeta(ref);
+	if (meta == NULL) {
+		lua_pushnil(L);
+		return 1;
+	}
+	lua_newtable(L);
+
+	{
```

Comment:
It doesn't produce overhead I just don't like seeing things for no reason because it can then cause somebody looking at the code to ask the question "is this there for a reason?"

If it's copying the style of something existing then maybe it's better to leave it as is